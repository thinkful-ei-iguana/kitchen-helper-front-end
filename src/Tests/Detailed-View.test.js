import React from 'react';
import ReactDOM from 'react-dom';
import DetailedView from '../Components/Detailed-View';
import { BrowserRouter } from 'react-router-dom';
import renderer from "react-test-renderer";

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter>
        <DetailedView />
    </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
    const tree = renderer.create(<BrowserRouter>
        <DetailedView />
    </BrowserRouter>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});