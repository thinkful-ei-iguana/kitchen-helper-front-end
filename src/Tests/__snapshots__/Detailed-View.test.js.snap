// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="view"
>
  <div
    className="image-container"
  >
    <div
      className="image"
      style={
        Object {
          "backgroundImage": "url(undefined)",
        }
      }
    />
  </div>
  <p
    className="recipe-title"
  >
    Title:
  </p>
  <h2
    className="recipe-title"
  />
  <p
    className="recipeDescription"
  >
    Recipe Description: 
  </p>
  <h3
    className="recipeDescription"
  />
  <p
    className="recipeIngredients"
  >
    Recipe Ingredients:
  </p>
  <h3
    className="recipeIngredients"
  />
  <p
    className="timeToMakeRecipe"
  >
    Time to make the recipe:
  </p>
  <h3
    className="timeToMakeRecipe"
  />
  <p
    className="recipeOwner"
  >
    Owner:
  </p>
  <h3
    className="recipeOwner"
  />
  <div
    className="recipe-date_created"
  >
    <span
      className="recipe-date_created"
    >
      Date Created: 
    </span>
  </div>
  <div>
    <div
      className="delete-recipe-button-div"
    >
      <button
        className="delete-recipe-button"
        onClick={[Function]}
      >
        Delete Recipe
      </button>
    </div>
  </div>
  <div>
    <div
      className="ownerSelectors"
    >
      <a
        className="editRecipe"
        href="/edit-recipe/undefined"
        onClick={[Function]}
      >
        Edit Recipe
      </a>
    </div>
  </div>
  <a
    href="/"
    onClick={[Function]}
  >
    <button
      className="cancel-view"
    >
      Cancel
    </button>
  </a>
</div>
`;
